src/Microsoft.DotNet.Watcher.Tools/Internal/MsBuildProjectFinder.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Watcher.Tools;
+
+namespace Microsoft.DotNet.Watcher.Internal
+{
+    internal class MsBuildProjectFinder
```

Comment:
This would be a good API to add to Microsoft.DotNet.Cli.Utils. The SDK itself needs this too.